editor/src/clj/editor/code/script_compilation.clj

```diff
@@ -0,0 +1,275 @@
+            (log/error :message log-error-message :exception exception)
+            (if-some [[proj-path line-number] (try-parse-file-line exception-message)]
+              (let [exception-resource (workspace/resolve-resource resource proj-path)
+                    exception-node-id (proj-path->resource-node proj-path)
+                    error-node-id (or exception-node-id _node-id)
+                    error-resource (if (nil? exception-node-id) resource exception-resource)
+                    error-user-data (r/make-code-error-user-data proj-path line-number)]
+                (g/->error error-node-id :modified-lines :fatal error-resource build-error-message error-user-data))
+              (g/->error _node-id :modified-lines :fatal resource build-error-message)))
+          (let [preprocessed-lua-info
+                (with-open [reader (data/lines-reader preprocessed-lines)]
+                  (lua-parser/lua-info workspace valid-resource-kind? reader))
+
+                preprocessed-script-properties (lua-info->script-properties preprocessed-lua-info)
+                preprocessed-modules (lua-info->modules preprocessed-lua-info)
+                preprocessed-go-props-with-source-resources
+                (map (fn [{:keys [name type value]}]
+                       (let [go-prop-type (script-property-type->go-prop-type type)
+                             go-prop-value (properties/clj-value->go-prop-value go-prop-type value)]
+                         {:id name
+                          :type go-prop-type
+                          :value go-prop-value
+                          :clj-value value}))
+                     preprocessed-script-properties)
+
+                proj-path->resource-property-build-target
+                (bt/make-proj-path->build-target original-resource-property-build-targets)
+
+                [preprocessed-go-props preprocessed-go-prop-dep-build-targets]
+                (properties/build-target-go-props proj-path->resource-property-build-target preprocessed-go-props-with-source-resources)]
+            ;; NOTE: The :user-data must not contain any overridden data. If it does,
+            ;; the build targets won't be fused and the script will be recompiled
+            ;; for every instance of the script component. The :go-props here describe
+            ;; the original property values from the script, never overridden values.
+            [(bt/with-content-hash
+               {:node-id _node-id
+                :resource (workspace/make-build-resource resource)
+                :build-fn build-script
+                :user-data {:lines preprocessed-lines
+                            :go-props preprocessed-go-props
+                            :modules preprocessed-modules
+                            :proj-path (resource/proj-path resource)}
+                :deps preprocessed-go-prop-dep-build-targets
+                :dynamic-deps (mapv lua/lua-module->path preprocessed-modules)})]))))))
```

Comment:
99% of the code in this file is taken from `script.clj`. There is a small behavior difference: module dependencies are now `:dynamic-deps` that only reference dependent file paths instead of including the dependent build targets.